akka-stream/src/main/scala/cursor.scala

```diff
@@ -15,13 +16,8 @@ import org.reactivestreams.Publisher
 import akka.stream.Materializer
 import akka.stream.scaladsl.{ Sink, Source }
 
-/** For future extension */
-sealed trait State {}
```

Comment:
For compatibility, keep as sealed trait with `apply` and `unapply` in companion

Reply:
I think this should not cause any incompatibility: after all, user code could not have extended that interface, nor could have pattern matched on it, nor call any method on it (there was no method).

Reply:
it indeed does, see that MiMa report.

Reply:
what I meant is that MiMa is only looking at the change `trait` -> `class` but in this particular instance (`sealed trait` with no methods) I believe it cannot induce backward incompatibilities in practice.
But anyway... I kept the `sealed` trait and introduced sub-`case` classes. MiMa is happy. (with some excludes for package private changes)